org.eclipse.jdt.ui/ui/org/eclipse/jdt/ui/actions/MakeStaticAction.java

```diff
@@ -0,0 +1,153 @@
+import org.eclipse.jdt.internal.ui.javaeditor.JavaTextSelection;
+import org.eclipse.jdt.internal.ui.refactoring.RefactoringMessages;
+import org.eclipse.jdt.internal.ui.util.ExceptionHandler;
+
+/**
+ * Extracts an expression into a constant field and replaces all occurrences of the expression with
+ * the new constant.
+ *
+ * Modifies a method with the static keyword and changes all method invocations to properly invoke
+ * the refactored method.
+ *
+ * @since 3.29
+ *
+ */
+public final class MakeStaticAction extends SelectionDispatchAction {
+
+	private JavaEditor fEditor;
+
+	/**
+	 * Note: This constructor is for internal use only. Clients should not call this constructor.
+	 *
+	 * @param editor the java editor. Must not be null.
+	 *
+	 * @noreference This constructor is not intended to be referenced by clients.
+	 */
+	public MakeStaticAction(JavaEditor editor) {
+		super(editor.getEditorSite());
+		setText(RefactoringMessages.MakeStaticAction_title);
+		setToolTipText(RefactoringMessages.MakeStaticAction_tooltip);
+		setDescription(RefactoringMessages.MakeStaticAction_description);
+		fEditor= editor;
+		setEnabled(SelectionConverter.getInputAsCompilationUnit(fEditor) != null);
+		PlatformUI.getWorkbench().getHelpSystem().setHelp(this, IJavaHelpContextIds.MAKE_STATIC_ACTION);
+	}
+
+	public MakeStaticAction(IWorkbenchSite site) {
+		super(site);
+		setText(RefactoringMessages.MakeStaticAction_title);
+		setToolTipText(RefactoringMessages.MakeStaticAction_tooltip);
+		setDescription(RefactoringMessages.MakeStaticAction_description);
+		PlatformUI.getWorkbench().getHelpSystem().setHelp(this, IJavaHelpContextIds.MAKE_STATIC_ACTION);
+	}
+
+	// Selected in Editor window
+	@Override
+	public void selectionChanged(ITextSelection selection) {
+		setText(RefactoringMessages.MakeStaticAction_title);
+		setToolTipText(RefactoringMessages.MakeStaticAction_tooltip);
+		setDescription(RefactoringMessages.MakeStaticAction_description);
+		setEnabled(SelectionConverter.getInputAsCompilationUnit(fEditor) != null);
+	}
+
+	// Selected in outline window
+	@Override
+	public void selectionChanged(IStructuredSelection selection) {
+		try {
+			setEnabled(RefactoringAvailabilityTester.isMakeStaticAvailable(selection));
+		} catch (JavaModelException e) {
+			if (JavaModelUtil.isExceptionToBeLogged(e))
+				JavaPlugin.log(e);
+		}
+	}
+
+	/**
+	 * Note: This method is for internal use only. Clients should not call this method.
+	 *
+	 * @param selection the Java text selection (internal type)
+	 *
+	 * @noreference This method is not intended to be referenced by clients.
+	 */
+	@Override
+	public void selectionChanged(JavaTextSelection selection) {
+		try {
+			setEnabled(RefactoringAvailabilityTester.isMakeStaticAvailable(selection));
+		} catch (JavaModelException e) {
+			setEnabled(false);
+		}
+	}
+
+	@Override
+	public void run(ITextSelection selection) {
+		if (!ActionUtil.isProcessable(fEditor))
+			return;
```

Comment:
Please check all statements that require an expression block (if, while, for etc.) for having braces around the expression block. Leaving out these braces should be avoided as it is quite dangerous.

If you don't trust me, take a look at the famous Apple "goto fail" SSL bug ;-)
https://www.spiegel.de/netzwelt/web/goto-fail-apples-furchtbarer-fehler-a-955154.html

Reply:
Done. :)